cdap-ui/app/cdap/components/Home/index.js

```diff
@@ -173,7 +259,57 @@ class Home extends Component {
     this.setState({selectedEntity: uniqueId});
   }
 
+  isDefaultFilter(){
+    return (this.state.filter.length === defaultFilter.length) && this.state.filter.every((element, index) => {
+      return element === defaultFilter[index];
+    });
+  }
+
+  isDefaultSort(){
+    return this.state.sortObj.sort === this.sortOptions[0].sort;
+  }
+
+  makeSortFilterParams(){
+    let sortAndFilterParams = '';
+    let filterString = '';
+    let isDefaultSorted = this.isDefaultSort();
+    let isDefaultFiltered = this.isDefaultFilter();
+    let searchTerm = this.state.query;
+
+    if(isDefaultSorted && isDefaultFiltered){
+      return;
+    }
+
+    // //Add Query Params to URL on re-render
+    sortAndFilterParams = !isDefaultSorted ? '?sort=' + this.state.sortObj.sort.split(' ').join('+') : '';
+
+    if(!isDefaultFiltered){
+      //If the cards are sorted and filtered, seperate query params
+      sortAndFilterParams = !isDefaultSorted ? sortAndFilterParams.concat('&') : '?';
+      filterString = this.state.filter.join(',');
+      sortAndFilterParams = sortAndFilterParams + 'filter=' + filterString;
+    }
+
+    if(searchTerm.length > 0){
+      if(sortAndFilterParams.length > 0){
+        sortAndFilterParams = sortAndFilterParams + '&search=' + searchTerm;
+      } else {
+        sortAndFilterParams = '?search=' + searchTerm;
+      }
+    }
+
+    let obj = {
+      title: 'CDAP',
+      url: location.pathname + sortAndFilterParams
+    };
+
+    history.pushState(obj, obj.Title, obj.url);
```

Comment:
also I believe this will cause problems with clicking back button. Have you discuss what is the expected behavior?